src/api/spec/models/workflow/step/link_package_step_spec.rb

```diff
@@ -0,0 +1,310 @@
+
+RSpec.describe Workflow::Step::LinkPackageStep, vcr: true do
+  let!(:user) { create(:confirmed_user, :with_home, login: 'Iggy') }
+  let(:token) { create(:workflow_token, user: user) }
+  let(:target_project_name) { "home:#{user.login}:#{project.name}:PR-1" }
+
+  subject do
+    described_class.new(step_instructions: step_instructions,
+                        scm_extractor_payload: scm_extractor_payload,
+                        token: token)
+  end
+
+  RSpec.shared_context 'source_project not provided' do
+    let(:step_instructions) { { source_package: package.name } }
+
+    it { expect { subject.call }.to(change(Package, :count).by(0)) }
+  end
+
+  RSpec.shared_context 'source_package not provided' do
+    let(:step_instructions) { { source_project: package.project.name } }
+
+    it { expect { subject.call }.to(change(Package, :count).by(0)) }
+  end
+
+  RSpec.shared_context 'failed when source_package does not exist' do
+    let(:step_instructions) do
+      {
+        source_project: project.name,
+        source_package: 'this_package_does_not_exist'
+      }
+    end
+
+    it { expect { subject.call }.to raise_error(Workflow::Step::Errors::CanNotLinkPackageNotFound) }
+  end
+
+  RSpec.shared_context 'project and package does not exist' do
+    let(:step_instructions) do
+      {
+        source_project: 'invalid project',
+        source_package: 'invalid package'
+      }
+    end
+
+    it { expect { subject.call }.to raise_error(Project::Errors::UnknownObjectError) }
+  end
+
+  RSpec.shared_context 'target package already exists' do
+    # Emulates that the target project and package already existed
+    let!(:linked_project) { create(:project, name: "home:#{user.login}:#{package.project.name}:PR-1", maintainer: user) }
+    let!(:linked_package) { create(:package_with_file, name: package.name, project: linked_project) }
+
+    let(:step_instructions) do
+      {
+        source_project: project.name,
+        source_package: package.name
+      }
+    end
+
+    it { expect { subject.call }.to raise_error(Workflow::Step::Errors::DuplicatePackageError) }
+  end
+
+  RSpec.shared_context 'failed without link permissions' do
+    let(:step_instructions) do
+      {
+        source_project: project.name,
+        source_package: package.name
+      }
+    end
+
+    before do
+      # rubocop:disable RSpec/AnyInstance
+      allow_any_instance_of(Package).to receive(:disabled_for?).with('sourceaccess', nil, nil).and_return(true)
+      # rubocop:enable RSpec/AnyInstance
+    end
+
+    it { expect { subject.call }.to raise_error(Pundit::NotAuthorizedError) }
+  end
+
+  RSpec.shared_context 'successful new PR or MR event' do
+    let(:step_instructions) do
+      {
+        source_project: package.project.name,
+        source_package: package.name
+      }
+    end
+
+    it { expect { subject.call }.to(change(Project, :count).by(1)) }
+    it { expect { subject.call }.to(change(Package, :count).by(2)) }
```

Comment:
Is it really creating 2 packages? I think it should be 1 package.

Reply:
It's creating two packages. The other one is [this](https://github.com/openSUSE/open-build-service/pull/11413/files#diff-b03f5a9fe996b716789616806c138c935118ab81fae1527c9bdcbd03739eb13bR61). This special package is needed to trigger the service.

Reply:
[Here](https://github.com/openSUSE/open-build-service/pull/11413/files#diff-b03f5a9fe996b716789616806c138c935118ab81fae1527c9bdcbd03739eb13bR62-R66) is the "patch" to solve the problem with the service.